vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+
+using namespace std;
+typedef float weight;
+
+class array_parameters
+{
+private:
+	weight* _begin;
+
+public:
+	array_parameters(weight* begin)
```

Comment:
Maybe this constructor should not exist?  It seems dangerous to allow external allocation.  I'd rather be absolutely positive that the mask matches the length of the array.